### PR TITLE
[Enhancement] Debug training process

### PR DESCRIPTION
**Pull request by shiyutang**

### CODE REVIEW

File: paddleseg3d/core/infer.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+        ori_shape (list): Origin shape of image.
+        transforms (list): List of transform.
+
+    Returns:
+        list: List of tuple, there are two format:
+            ('resize', (h, w)) The image shape before resize,
+            ('padding', (h, w)) The image shape before padding.
+    """
+    reverse_list = []
+    h, w = ori_shape[0], ori_shape[1]
```
Comment (parap1uie-s) [inline paddleseg3d/core/infer.py]: H, W for 2D Image. 3D <==> D, H, W,

File: paddleseg3d/core/infer.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    for op in transforms:
+        if op.__class__.__name__ in ['Resize']:
+            reverse_list.append(('resize', (h, w)))
```
Comment (parap1uie-s) [inline paddleseg3d/core/infer.py]: check

File: paddleseg3d/core/infer.py [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+            h, w = op.target_size[0], op.target_size[1]
+        if op.__class__.__name__ in ['ResizeByLong']:
```
Comment (parap1uie-s) [inline paddleseg3d/core/infer.py]: Consider remove this segment for simplicity

File: paddleseg3d/core/infer.py [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+            short_edge = min(h, w)
+            short_edge = int(round(short_edge * op.long_size / long_edge))
+            long_edge = op.long_size
+            if h > w:
+                h = long_edge
+                w = short_edge
+            else:
+                w = long_edge
+                h = short_edge
+        if op.__class__.__name__ in ['ResizeByShort']:
```
Comment (parap1uie-s) [inline paddleseg3d/core/infer.py]: Same as above

File: paddleseg3d/core/infer.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    cols = np.int(np.ceil(1.0 * (w_im - w_crop) / w_stride)) + 1
+    # prevent negative sliding rounds when imgs after scaling << crop_size
+    rows = 1 if h_im <= h_crop else rows
+    cols = 1 if w_im <= w_crop else cols
+    # TODO 'Tensor' object does not support item assignment. If support, use tensor to calculation.
+    final_logit = None
+    count = np.zeros([1, 1, h_im, w_im])
+    for r in range(rows):
+        for c in range(cols):
+            h1 = r * h_stride
```
Comment (parap1uie-s) [inline paddleseg3d/core/infer.py]: H, W, ?

File: paddleseg3d/datasets/lung_coronavirus.py [hunk 6]
```diff
@@ -101,8 +102,25 @@ def __init__(self,
         result_dir="data/lung_coronavirus/lung_coronavirus_phase1",
         transforms=[],
         mode="train")
+    import paddle
```
Comment (parap1uie-s) [inline paddleseg3d/datasets/lung_coronavirus.py]: `import` should not exists here

File: paddleseg3d/datasets/lung_coronavirus.py [hunk 7]
```diff
@@ -101,8 +102,25 @@ def __init__(self,
...
+        dataset,
+        batch_sampler=batch_sampler,
+        num_workers=2,
+        return_list=True,
+        worker_init_fn=worker_init_fn,
+    )
+
+    for data in loader:
+        img, label = data
         print(img.shape, label.shape)
```
Comment (parap1uie-s) [inline paddleseg3d/datasets/lung_coronavirus.py]: delete

**Review by parap1uie-s — CHANGES_REQUESTED** [review]

File: paddleseg3d/datasets/lung_coronavirus.py (hunk 7 above)
Comment (shiyutang) [reply paddleseg3d/datasets/lung_coronavirus.py]: done

File: paddleseg3d/datasets/lung_coronavirus.py (hunk 6 above)
Comment (shiyutang) [reply paddleseg3d/datasets/lung_coronavirus.py]: done

File: paddleseg3d/core/infer.py (hunk 4 above)
Comment (shiyutang) [reply paddleseg3d/core/infer.py]: removed, need to add by ourselves

File: paddleseg3d/core/infer.py (hunk 3 above)
Comment (shiyutang) [reply paddleseg3d/core/infer.py]: removed, need to add by ourselves

File: paddleseg3d/core/infer.py (hunk 2 above)
Comment (shiyutang) [reply paddleseg3d/core/infer.py]: removed, need to add by ourselves

File: paddleseg3d/core/infer.py (hunk 1 above)
Comment (shiyutang) [reply paddleseg3d/core/infer.py]: done

File: paddleseg3d/core/infer.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        ori_shape (list): Origin shape of image.
+        transforms (list): List of transform.
+
+    Returns:
+        list: List of tuple, there are two format:
+            ('resize', (h, w)) The image shape before resize,
+            ('padding', (h, w)) The image shape before padding.
+    """
+    reverse_list = []
+    d, h, w = ori_shape[0], ori_shape[1], ori_shape[3]
```
Comment (parap1uie-s) [inline paddleseg3d/core/infer.py]: 0, 1, 3?

File: paddleseg3d/core/train.py [hunk 9]
```diff
@@ -171,11 +171,11 @@ def train(model,
...
-                edges = data[2].astype('int64')
-            if hasattr(model, 'data_format') and model.data_format == 'NHWC':
-                images = images.transpose((0, 2, 3, 1))
+
+            if hasattr(
+                    model, 'data_format'
+            ) and model.data_format == 'NDHWC':  # originally as NCDHW
+                images = images.transpose((0, 2, 3, 4, 1))
 
             if fp16:
```
Comment (parap1uie-s) [inline paddleseg3d/core/train.py]: https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/amp/Overview_cn.html

conv3d等算子不支持fp16模式吧？这个flag有没有意义

File: paddleseg3d/core/infer.py (hunk 8 above)
Comment (shiyutang) [reply paddleseg3d/core/infer.py]: done

File: paddleseg3d/core/train.py (hunk 9 above)
Comment (shiyutang) [reply paddleseg3d/core/train.py]: deleted~

**Review by parap1uie-s — APPROVED** [review]
LGTM

**Pull request merged**